result.go

```diff
@@ -96,6 +100,19 @@ func newResultList(ctype reflect.Type) (resultList, error) {
 
 func (rl resultList) Produces() map[key]struct{} { return rl.produces }
 
+func (resultList) Extract(*Container, reflect.Value) error {
+	panic("resultList.Extract() must never be called")
+}
+
+func (rl resultList) ExtractResults(c *Container, values []reflect.Value) error {
```

Comment:
`ExtractAll` or `ExtractList`? These are all `result` implementations, so `Extract` and `ExtractResults` aren't obviously different to me.